x-pack/plugins/aiops/public/components/log_categorization/category_table/category_table.tsx

```diff
@@ -302,49 +238,49 @@ export const CategoryTable: FC<Props> = ({
     };
   };
 
-  return (
-    <>
-      <TableHeader
-        categoriesCount={categories.length}
-        selectedCategoriesCount={selectedCategories.length}
-        labels={labels}
-        openInDiscover={(queryMode: QueryMode) => openInDiscover(queryMode)}
-      />
-      <EuiSpacer size="xs" />
-      <EuiHorizontalRule margin="none" />
+  const tableStyle = css({
+    thead: {
+      position: 'sticky',
+      insetBlockStart: 0,
+      zIndex: 1,
+      backgroundColor: euiTheme.euiColorEmptyShade,
+      boxShadow: `inset 0 0px 0, inset 0 -1px 0 ${euiTheme.euiBorderColor}`,
+    },
+  });
 
-      <EuiInMemoryTable<Category>
-        compressed
-        items={categories}
-        columns={columns}
-        selection={selectionValue}
-        itemId="key"
-        onTableChange={onTableChange}
-        pagination={pagination}
-        sorting={sorting}
-        data-test-subj="aiopsLogPatternsTable"
-        itemIdToExpandedRowMap={itemIdToExpandedRowMap}
-        rowProps={(category) => {
-          return enableRowActions
-            ? {
-                onClick: () => {
-                  if (category.key === pinnedCategory?.key) {
-                    setPinnedCategory(null);
-                  } else {
-                    setPinnedCategory(category);
-                  }
-                },
-                onMouseEnter: () => {
-                  setSelectedCategory(category);
-                },
-                onMouseLeave: () => {
-                  setSelectedCategory(null);
-                },
-                style: getRowStyle(category),
-              }
-            : undefined;
-        }}
-      />
-    </>
+  return (
+    <EuiInMemoryTable<Category>
+      compressed
+      items={categories}
+      columns={columns}
```

Comment:
What do you think about including the name of the selected categorization field in the column heading e.g. `error.message examples`? Otherwise the only way to know which field is being used is by opening the Options popover. Would also be helpful in the warning callout and expanded row:

<img width="583" alt="Screenshot 2024-05-15 at 12 45 03" src="https://github.com/elastic/kibana/assets/7405507/acadc081-be91-4d27-9d52-92c26ac910ca">

Reply:
Updated in https://github.com/elastic/kibana/pull/178916/commits/820b2be0f5ff2884410b273a47eb996f47d6a374
![image](https://github.com/elastic/kibana/assets/22172091/c68b21e1-606d-4ac2-bcc3-34b2266afd57)

Reply:
I'm thinking now that I should move the field selection out of the menu and have it at the top. Clicking it would open a popover with the field select.
<img width="411" alt="image" src="https://github.com/elastic/kibana/assets/22172091/e2aadf29-4f85-4973-a35b-57e8204dc5f1">

Reply:
I like the separate control for selecting the field.